x-pack/plugins/ingest_pipelines/public/application/pipeline_editor/components/processor_settings_form/processors/common_fields/processor_type_field.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+import { EuiComboBox } from '@elastic/eui';
+import React, { FunctionComponent } from 'react';
+import {
+  FIELD_TYPES,
+  FieldConfig,
+  UseField,
+  fieldValidators,
+  FormRow,
+} from '../../../../../../shared_imports';
+import { types } from '../../map_processor_type_to_form';
+
+interface Props {
+  initialType?: string;
+}
+
+const { emptyField } = fieldValidators;
+
+const typeConfig: FieldConfig = {
+  type: FIELD_TYPES.TEXT,
+  label: i18n.translate('xpack.ingestPipelines.pipelineEditor.typeField.typeFieldLabel', {
+    defaultMessage: 'Type',
+  }),
+  validations: [
+    {
+      validator: emptyField(
+        i18n.translate('xpack.ingestPipelines.pipelineEditor.typeField.fieldRequiredError', {
+          defaultMessage: 'A type is required.',
+        })
+      ),
+    },
+  ],
+};
+
+export const ProcessorTypeField: FunctionComponent<Props> = ({ initialType }) => {
+  return (
+    <FormRow
+      title={i18n.translate('xpack.ingestPipelines.pipelineEditor.typeFieldTitle', {
+        defaultMessage: 'Type',
+      })}
+    >
+      <UseField config={typeConfig} path={'type'} defaultValue={initialType}>
+        {typeField => {
+          return (
+            <EuiComboBox
```

Comment:
There is an `<EuiComboBoxField />` that already takes care of serialization, deserialization, validation at the array level, validation at the item level. I would recommend using it.

This is the config (from the schema)

https://github.com/elastic/kibana/blob/master/x-pack/plugins/index_management/public/application/components/template_form/template_form_schemas.tsx#L100

You see 2 validations, at the Array level (the first one), and at the item level (the second one).

And this is how it is simply used

https://github.com/elastic/kibana/blob/master/x-pack/plugins/index_management/public/application/components/template_form/steps/step_logistics.tsx#L127

No need to re-write the logic of `onChange`, `options`, `selectedOptions`